nsxt/resource_nsxt_manager_cluster.go

```diff
@@ -175,75 +132,58 @@ func getClusterInfoFromHostNode(d *schema.ResourceData, m interface{}) (string,
 	min := c.CommonConfig.MinRetryInterval
 	max := c.CommonConfig.MaxRetryInterval
 	maxRetries := c.CommonConfig.MaxRetries
-	hostIP := ""
+	hostIPs := []string{}
 	for i := 0; i < maxRetries; i++ {
 		clusterConfig, err := client.Get()
 		if err != nil {
-			return "", "", "", handleReadError(d, "Cluster Config", "", err)
+			return "", "", hostIPs, err
 		}
-		if hostIP == "" {
-			hostIP, err = getHostIPFromClusterConfig(m, clusterConfig)
+		if len(hostIPs) == 0 {
+			hostIPs, err = resolveHostIPs(m)
 			if err != nil {
-				return "", "", "", err
+				return "", "", hostIPs, err
 			}
+
+			log.Printf("[DEBUG]: Host resolved to IP addresses %v", hostIPs)
 		}
 		clusterID := *clusterConfig.ClusterId
 		nodes := clusterConfig.Nodes
 		node := nodes[0]
 		apiListenAddr := node.ApiListenAddr
 		if apiListenAddr != nil {
```

Comment:
Could it be better to first do this check, and only if it succeeds collect hostIPs? It seems while we wait for ApiListenAddr to populate we will keep repeating `resolveHostIPs`

Reply:
`resolveHostIPs` only happens on first iteration, if `len(hostIPs) == 0`, so it shouldn't be repeated..